src/components/outline-nav/components/outline-link-with-nesting/index.tsx

```diff
@@ -0,0 +1,25 @@
+import { OutlineListItems, OutlineLink } from '..'
+import type { OutlineLinkWithNestingProps } from '../../types'
+import s from './outline-link-with-nesting.module.css'
+
+/**
+ * Render an outline nav link,
+ * with additional outline links indented in a nested section.
+ */
+function OutlineLinkWithNesting({
```

Comment:
❓  Have we considered limiting the levels of nesting allowed?

Reply:
In a way, yes! We considered writing code-level checks, such as throwing an error where nesting is too deep. However, this felt a little strange to do at this point, since the only way nesting could become too deep is if we go in and change code to generate deeply nested items and then pass them to `OutlineNav`.

Personally my hot take is: code-level protections seem most useful when dealing with authored content, where the limitations might not be clear, and won't have to go through code review to be bypassed. If we expose `OutlineNav` in a way where authors can somehow generate the levels of nesting, then perhaps code-level protection would be useful... however even then, I think we'd probably lean more in the "content conformance" direction, at the "extract" part of ETL rather than during rendering 🤔